core/src/main/java/org/apache/calcite/sql/fun/SqlPositionFunction.java

```diff
@@ -88,6 +95,10 @@ public SqlPositionFunction(String name) {
       return OperandTypes.SAME_SAME_INTEGER.checkOperandTypes(
           callBinding, throwOnFailure)
           && super.checkOperandTypes(callBinding, throwOnFailure);
+    case 4:
+          return OperandTypes.sequence("INSTR(<STRING>, <STRING>, <INTEGER>, <INTEGER>)", OperandTypes.STRING, OperandTypes.STRING, OperandTypes.INTEGER, OperandTypes.INTEGER).checkOperandTypes(
```

Comment:
So this is the alternative to adding a SAME_SAME_INTEGER_INTEGER entry?

Reply:
yeah, essentially will just check that the first 2 are in string family, but won't check them against each other. Will error later on when it realizes there's no matching function call to INSTR() for one Bytestring one String